examples/upgrade-http1x.md

```diff
@@ -0,0 +1,645 @@
+```
+
+**After:**
+```toml
+[dependencies]
+aws-smithy-http-client = "1.0"  # Replaces direct hyper usage
+aws-smithy-runtime = { version = "1.0", features = ["client"] }
+aws-smithy-runtime-api = "1.0"
+```
+
+## Server-Side Changes
+
+### 1. Import Changes
+
+**Before:**
+```rust
+use hyper::StatusCode;
+```
+
+**After:**
+```rust
+use http::StatusCode;
+use pokemon_service_server_sdk::server::serve;
+use tokio::net::TcpListener;
+```
+
+**Key Points:**
+- `StatusCode` and other HTTP types now come from the `http` crate, not `hyper`
+- Import the `serve` helper from your server SDK
+- Use `tokio::net::TcpListener` instead of hyper's built-in listener
+
+### 2. Server Initialization
+
+**Before (hyper@0.14):**
+```rust
+#[tokio::main]
+pub async fn main() {
+    // ... setup config and app ...
+
+    let make_app = app.into_make_service_with_connect_info::<SocketAddr>();
+
+    let bind: SocketAddr = format!("{}:{}", args.address, args.port)
+        .parse()
+        .expect("unable to parse the server bind address and port");
+
+    let server = hyper::Server::bind(&bind).serve(make_app);
+
+    if let Err(err) = server.await {
+        eprintln!("server error: {err}");
+    }
+}
+```
+
+**After (hyper@1.x):**
+```rust
+#[tokio::main]
+pub async fn main() {
+    // ... setup config and app ...
+
+    let make_app = app.into_make_service_with_connect_info::<SocketAddr>();
+
+    let bind: SocketAddr = format!("{}:{}", args.address, args.port)
+        .parse()
+        .expect("unable to parse the server bind address and port");
+
+    let listener = TcpListener::bind(bind)
+        .await
+        .expect("failed to bind TCP listener");
+
+    // Optional: Get the actual bound address (useful for port 0)
+    let actual_addr = listener.local_addr().expect("failed to get local address");
+    eprintln!("Server listening on {}", actual_addr);
+
+    if let Err(err) = serve(listener, make_app).await {
+        eprintln!("server error: {err}");
+    }
+}
+```
+
+**Key Changes:**
+1. Replace `hyper::Server::bind(&bind)` with `TcpListener::bind(bind).await`
+2. Use the `serve()` helper function instead of `.serve(make_app)`
+3. TcpListener binding is now async (requires `.await`)
+4. Can get actual bound address with `.local_addr()` (useful for testing with port 0)
+
+### 3. Service Building
+
+The service building API remains the same:
+
+```rust
+let app = PokemonService::builder(config)
+    .get_pokemon_species(get_pokemon_species)
+    .get_storage(get_storage_with_local_approved)
+    .get_server_statistics(get_server_statistics)
+    .capture_pokemon(capture_pokemon)
+    .do_nothing(do_nothing_but_log_request_ids)
+    .check_health(check_health)
+    .build()
+    .expect("failed to build an instance of PokemonService");
+
+let make_app = app.into_make_service_with_connect_info::<SocketAddr>();
+```
+
+No changes needed here! The service builder API is stable.
+
+## Client-Side Changes
+
+### HTTP Client Connector Setup
```

Comment:
need to call out that client/server http versions are totally decoupled.

Reply:
I need to check with the SDK team on this section. For the time being I am removing it.